pkg/aggregate/aggregate_test.go

```diff
@@ -1,20 +1,81 @@
+// +build unit
+
 package aggregate
 
 import (
 	"fmt"
 	"testing"
 )
 
-func TestAggragators(t *testing.T) {
-	aggrString := "*"
-	aggr, err := AggrsFromString(aggrString)
-	if err != nil {
-		t.Fatal(err)
+func TestAggregators(t *testing.T) {
+	testCases := []struct {
+		desc               string
+		aggString          string
+		data               map[int64]float64
+		exprCol            string
+		bucket             int
+		expectedUpdateExpr string
+		expectedSetExpr    string
+		expectFail         bool
+	}{
+		{"Should aggregate data with Count aggregator", "count", map[int64]float64{1: 7.5, 2: 2.5}, "v", 1,
+			"_v_count[1]=_v_count[1]+2;", "_v_count[1]=2;", false},
+
+		{"Should aggregate data with Sum aggregator", "sum", map[int64]float64{1: 7.5, 2: 2.5}, "v", 1,
+			fmt.Sprintf("_v_sum[1]=_v_sum[1]+%f;", 10.0),
+			fmt.Sprintf("_v_sum[1]=%f;", 10.0), false},
+
+		{"Should aggregate data with Sqr aggregator", "sqr", map[int64]float64{1: 2.0}, "v", 1,
+			fmt.Sprintf("_v_sqr[1]=_v_sqr[1]+%f;", 4.0),
+			fmt.Sprintf("_v_sqr[1]=%f;", 4.0), false},
+
+		// todo: enable when bug is fixed - IG-8675
+		//{"Should aggregate data with Min & Max aggregators", "min,max", map[int64]float64{1: 7.5, 2: 2.5}, "v", 1,
+		//	fmt.Sprintf("_v_min[1]=min(_v_min[1],%f);_v_max[1]=max(_v_max[1],%f);", 2.5, 7.5),
+		//	fmt.Sprintf("_v_min[1]=%f;_v_max[1]=%f;", 2.5, 7.5), false},
```

Comment:
This pattern kinda sucks if we have to comment out tests and log them ourselves... I prefer good old code sharing by function (where the tests call the common code and not vice versa).